### PR TITLE
BUG: add minor version to MacOS 11 and 12 tags as workaround

### PR DESCRIPTION
**Pull request by FFY00**
Signed-off-by: Filipe Laíns <lains@riseup.net>

### CODE REVIEW

File: mesonpy/__init__.py [hunk 1]
```diff
@@ -672,7 +672,19 @@ def platform_tag(self) -> str:
             return 'any'
         # XXX: Choose the sysconfig platform here and let something like auditwheel
         #      fix it later if there are system dependencies (eg. replace it with a manylinux tag)
-        return sysconfig.get_platform().replace('-', '_').replace('.', '_')
+        platform_ = sysconfig.get_platform()
+        parts = platform_.split('-')
+        if parts[0] == 'macosx' and parts[1] in ('11', '12'):
+            # Workaround for bug where pypa/packaging does not consider Mac tags
+            # without minor versions valid. Mac started to do this in version
+            # 11, and pypa/packaging should handle things correctly from version
```
Comment (rgommers) [inline mesonpy/__init__.py:680]: I'd suggest adjusting this comment to say "Some Python flavors (Homebrew for example) on macOS started to do this in version 11".

**Review by rgommers — APPROVED** [review]
This code LTGM and works with SciPy and conda-forge Python, thanks @FFY00. I just suggested a change to one comment for clarity/correctness, because it's not what macOS does. My Python version for example:
```
>>> import sysconfig
>>> platform_ = sysconfig.get_platform()
>>> parts = platform_.split("-")
>>> parts
['macosx', '11.0', 'arm64']
```

File: mesonpy/__init__.py (hunk 1 above)
Comment (FFY00) [reply mesonpy/__init__.py]: Added :blush:

**Comment by FFY00** [comment]
Right, I have changed the comment accordingly. Thanks!

**Pull request merged**